robot/web/web_c.go

```diff
@@ -94,140 +92,54 @@ func (svc *webService) addNewStreams(ctx context.Context) error {
 			Name: name,
 		}
-
-		if isVideo {
-			config.VideoEncoderFactory = svc.opts.streamConfig.VideoEncoderFactory
-		} else {
-			config.AudioEncoderFactory = svc.opts.streamConfig.AudioEncoderFactory
-		}
-		stream, err := svc.streamServer.Server.NewStream(config)
-
-		// Skip if stream is already registered, otherwise raise any other errors
-		var registeredError *webstream.StreamAlreadyRegisteredError
-		if errors.As(err, &registeredError) {
-			return nil, true, nil
-		} else if err != nil {
-			return nil, false, err
-		}
-
-		if !svc.streamServer.HasStreams {
-			svc.streamServer.HasStreams = true
-		}
-		return stream, false, nil
-	}
-
-	for name, source := range svc.videoSources {
-		const isVideo = true
-		stream, alreadyRegistered, err := newStream(name, isVideo)
+		config.VideoEncoderFactory = svc.opts.streamConfig.VideoEncoderFactory
+		// TODO(RSDK-9079) Add reliable framerate fetcher for stream videosources
+		stream, err := svc.createStream(config, name)
 		if err != nil {
 			return err
-		} else if alreadyRegistered {
-			continue
 		}
-
-		svc.startVideoStream(ctx, source, stream)
+		svc.startVideoStream(ctx, svc.videoSources[name], stream)
 	}
-
-	for name, source := range svc.audioSources {
-		const isVideo = false
-		stream, alreadyRegistered, err := newStream(name, isVideo)
+	for name := range svc.audioSources {
+		config := gostream.StreamConfig{
+			Name: name,
+		}
+		config.AudioEncoderFactory = svc.opts.streamConfig.AudioEncoderFactory
+		stream, err := svc.createStream(config, name)
 		if err != nil {
 			return err
-		} else if alreadyRegistered {
-			continue
 		}
-
-		svc.startAudioStream(ctx, source, stream)
+		svc.startAudioStream(ctx, svc.audioSources[name], stream)
 	}
-
 	return nil
 }
 
-func (svc *webService) makeStreamServer(ctx context.Context) (*StreamServer, error) {
-	svc.refreshVideoSources()
-	svc.refreshAudioSources()
-	var streams []gostream.Stream
-	var streamTypes []bool
-
-	if svc.opts.streamConfig == nil || (len(svc.videoSources) == 0 && len(svc.audioSources) == 0) {
-		if len(svc.videoSources) != 0 || len(svc.audioSources) != 0 {
-			svc.logger.Debug("not starting streams due to no stream config being set")
-		}
-		noopServer, err := webstream.NewServer(streams, svc.r, svc.logger.Sublogger("stream"))
-		return &StreamServer{noopServer, false}, err
-	}
-
-	addStream := func(streams []gostream.Stream, name string, isVideo bool) ([]gostream.Stream, error) {
-		config := gostream.StreamConfig{
-			Name: name,
-		}
-		if isVideo {
-			config.VideoEncoderFactory = svc.opts.streamConfig.VideoEncoderFactory
-
-			// set TargetFrameRate to the framerate of the video source if available
-			props, err := svc.videoSources[name].MediaProperties(ctx)
-			if err != nil {
-				svc.logger.Warnw("failed to get video source properties", "name", name, "error", err)
-			} else if props.FrameRate > 0.0 {
-				// round float up to nearest int
-				config.TargetFrameRate = int(math.Ceil(float64(props.FrameRate)))
```

Comment:
Removed framerate prop passthrough to TargetFrameRate encoders settings for now.
- This was not being checked for reconfigures, only on initialization.
- After testing different cameras, this was only being triggered for Linux webcams.
  - Throttling stream by TargetFrameRate was not even helping as the stream.Next() for webcams waits for the next frame anyhow.
  
I can add this back in f we want, but created a ticket and TODO comment to address this in a more holistic way for all cameras. We will want reliable access to the framerate for bitrate calculations from different dynamic resolution options.

Reply:
This is the only code path we're removing right? All the other changes basically just change the stream registration ownership from `server.go` to `web_c.go?

Reply:
>  change the stream registration ownership from server.go to web_c.go?
This does not change ownership here. Step 2 after this PR is in will be moving video and audio sources into from `web_c.go` to `server.go`.